ksql-examples/src/main/java/io/confluent/ksql/datagen/TokenBucket.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.datagen;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Objects;
+
+public class TokenBucket {
```

Comment:
Why not use the Guava `RateLimiter`? https://guava.dev/releases/19.0/api/docs/index.html?com/google/common/util/concurrent/RateLimiter.html

Reply:
No good reason. I threw this together without much thought like a year ago to do ksql benchmarking. I agree it's better to just use that.